lib/perl/Genome/Process.pm

```diff
@@ -630,5 +633,97 @@ sub result_is_on_cle_disk_group {
     return Genome::Disk::Group::Validate::GenomeDiskGroups::is_cle_disk_group_name($disk_group_name);
 }
 
+sub compare_output {
+    my ($self, $other_process_id) = @_;
+    my $process_id = $self->id;
+
+    unless (defined ($other_process_id)) {
+        die $self->error_message('Require process ID argument!');
+    }
+    my $other_process = Genome::Process->get($other_process_id);
+    unless ($other_process) {
+        die $self->error_message('Could not get process for ID (%s)', $other_process_id);
+    }
+
+    unless ($self->class eq $other_process->class) {
+        die $self->error_message('Processes (%s) and (%s) are not of the same type', $process_id, $other_process_id);
+    }
+
+    my %diffs = $self->_compare_output_files($other_process);
+
+    return %diffs;
+}
+
+sub _compare_output_files {
+    my ($self, $other_process) = @_;
+
+    my $output_dir = Genome::Sys->create_temp_file_path();
+    my $other_output_dir = Genome::Sys->create_temp_file_path();
+
+    $self->symlink_results($output_dir);
+    $other_process->symlink_results($other_output_dir);
+
+    return $self->_compare_output_directories($output_dir, $other_output_dir, $other_process);
+}
+
+sub _compare_output_directories {
+    my ($self, $output_dir, $other_output_dir, $other_process) = @_;
+
+    my %diffs;
+    File::DirCompare->compare(
+        $other_output_dir,
+        $output_dir,
+        sub {
+            my ($other_file, $file) = @_;
+            if (! $file) {
+                if (-f $other_file) {
+                    $diffs{abs_path($other_file)} = sprintf(
+                        'no file %s found for process %s',
+                        File::Spec->abs2rel($other_file, $other_output_dir), $self->id
+                    );
+                }
+                elsif (-d $other_file) {
+                    $diffs{File::Spec->abs2rel($other_file, $other_output_dir)} = sprintf(
+                        'no directory %s found for process %s',
+                        File::Spec->abs2rel($other_file, $other_output_dir), $self->id
+                    );
+                }
+            } elsif (! $other_file) {
+                if (-f $file) {
+                    $diffs{abs_path($file)} = sprintf(
+                        'no file %s found for process %s',
+                        File::Spec->abs2rel($file, $output_dir), $other_process->id
+                    );
+                }
+                elsif (-d $file) {
+                    $diffs{File::Spec->abs2rel($file, $output_dir)} = sprintf(
+                        'no directory %s found for process %s',
+                        File::Spec->abs2rel($other_file, $other_output_dir), $self->id
+                    );
+                }
+            } else {
+                #Resolve possible symlinks
+                my $other_target = abs_path($other_file);
+                my $target = abs_path($file);
```

Comment:
It looks like `abs_path` will return empty string when an intermediate directory does not actually exist.  It only relates to symlinks in that the rule applies to the symlink's target.

Here's an example of a non-existant intermediate directory in path,

```
$ perl -e 'use strict; use warnings; use v5.10; use Cwd qw(abs_path); say q(real path = ) . abs_path(q(/Users)); say q(unreal file = ) . abs_path(q(/Users/foo)); say q(unreal directory (foo) = ) . abs_path(q(/Users/foo/bar))'
real path = /Users
unreal file = /Users/foo
unreal directory (foo) = 
```

So a symlink can be broken and still return an `abs_path` if it follow this rule, i.e. a symlink pointing to `/Users/foo` returns `/Users/foo` but a symlink pointing to `/Users/foo/bar` return an empty string.